glommio/benches/nop.rs

```diff
@@ -0,0 +1,99 @@
+//! Benchmark the noop performance of Glommio. This is a stress test for the executor/reactor
+//! infrastructure.
+
+use glommio::{LocalExecutorBuilder, Task};
+use std::fmt;
+use std::time::{Duration, Instant};
+
+struct Bench {
+    num_tasks: u64,
+    num_events: u64,
+}
+
+const BENCH_RUNS: &'static [Bench] = &[
+    Bench {
+        num_tasks: 100,
+        num_events: 10_000_000,
+    },
+    Bench {
+        num_tasks: 1_000,
+        num_events: 10_000_000,
+    },
+    Bench {
+        num_tasks: 10_000,
+        num_events: 10_000_000,
+    },
+    Bench {
+        num_tasks: 100_000,
+        num_events: 10_000_000,
+    },
+    Bench {
+        num_tasks: 1_000_000,
+        num_events: 10_000_000,
+    },
+];
+
+struct Measurement {
+    total_events: u64,
+    total_tasks: u64,
+    duration: Duration,
+}
+
+impl fmt::Display for Measurement {
```

Comment:
not a request for change, but I am wondering if this code couldn't be made generic for the other benchmarks.
The creation of the executor is potentially generic as well. We'd have a facility to create a measuring executor (where we pin to CPU0, and count the loops + time) and a helper executor (where we pin to some other CPU, and don't count).

See the TCP benchmarks or shared channel to see how those would be handy.